ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh

```diff
@@ -117,118 +144,209 @@ metadata:
+          After=migrate-to-local-etcd.service
+          Before=crio.service
+ 
+          # Only if /var/lib/etcd is mounted
+          Requisite=var-lib-etcd.mount
+ 
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+ 
+          # Do a quick check of the mountpoint directory before doing a full recursive relabel
+          # If restorecon /var/lib/etcd would not relabel the directory, don't
+          # run the recursive relabel
+          ExecCondition=/bin/bash -c "[ -n \"$(restorecon -nv /var/lib/etcd)\" ]"
+ 
+          ExecStart=/usr/sbin/restorecon -R /var/lib/etcd
+ 
           [Install]
-          WantedBy=local-fs.target
+          RequiredBy=var-lib-etcd.mount
         enabled: true
-        name: var-lib-etcd.mount
+        name: relabel-var-lib-etcd.service
 EOF
+info "INFO: Applying 98-var-lib-etcd MachineConfig"
+oc create -f /tmp/98-var-lib-etcd.yaml
+
+# Verify:
+# 
+# - The cluster is operating correctly
+# - /var/lib/etcd is a new mount
+# - /var/lib/etcd has the correct SELinux labels
+wait_for_machineconfig_done
+check_etcd_mounted_on_local_disk
+info "INFO: Moving etcd on local disk is complete and the cluster is healthy!"
+
+if [[ "${ETCD_ON_LOCAL_DISK_FULL_TEST}" == "false" ]]; then
+    info "INFO: Test is complete, exiting..."
+    exit 0
+fi
+
+info "This tests that the MachineConfig operates correctly when configured on a machine with no ephemeral disk."
+# Preparation:
+#
+# - Update CPMS to remove local etcd
+# - Wait for rollout to complete
+info "INFO: Remove additionalBlockDevices to the control plane machineset"
+oc patch ControlPlaneMachineSet/cluster -n openshift-machine-api --type json -p '[{"op": "remove", "path": "/spec/template/machines_v1beta1_machine_openshift_io/spec/providerSpec/value/additionalBlockDevices"}]'
+wait_for_cpms_done
 
+# Verify:
+# 
+# - The cluster is operating correctly
+# - The etcd block device is not present on the control plane machines
+# - There will be a failed systemd service: create-local-etcd.service
 wait_for_machineconfig_done
+check_etcd_unmounted_on_local_disk
+cp_machines=$(oc get machines -n openshift-machine-api --selector='machine.openshift.io/cluster-api-machine-role=master' --no-headers -o custom-columns=NAME:.metadata.name)
+for machine in ${cp_machines}; do
+  if oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value}' | grep -q 'additionalBlockDevices'; then
+    info "ERROR: Machine ${machine} does has the etcd block device"
+    exit 1
+  fi
+  info "INFO: Machine ${machine} does not have the etcd block device"
+done
+info 'INFO: All control plane machines do not have the etcd block device'
+cp_nodes=$(oc get nodes -l node-role.kubernetes.io/master --no-headers -o custom-columns=NAME:.metadata.name)
+for node in ${cp_nodes}; do
+  info "INFO: Checking create-local-etcd.service on node ${node}"
+  service=$(oc debug node/"${node}" -- chroot /host systemctl status create-local-etcd.service)
```

Comment:
I think it would be better to test for the absence of /dev/vdb here. This is really a different test. I probably wouldn't key the test of implementation details of the code like the name of the systemd unit if at all possible.